sky/resources.py

```diff
@@ -168,6 +168,9 @@ def __init__(
         self._is_image_managed = _is_image_managed
 
         self._disk_tier = disk_tier
+        if ports is not None:
+            if not isinstance(ports, list):
+                ports = [ports]
```

Comment:
Should we convert `self._ports` to `Optional[List[str]]` only, so as to simplify the types for internal codes?